fields/types/number/NumberFilter.js

```diff
@@ -10,92 +10,108 @@ const MODE_OPTIONS = [
 	{ label: 'Between', value: 'between' },
 ];
 
-var NumberFilter = React.createClass({
+function getDefaultValue () {
+	return {
+		mode: MODE_OPTIONS[0].value,
+		value: '',
+	};
+}
 
-	getInitialState () {
+var NumberFilter = React.createClass({
+	statics: {
+		getDefaultValue: getDefaultValue,
+	},
+	getDefaultProps () {
 		return {
-			modeValue: MODE_OPTIONS[0].value, // 'matches'
-			modeLabel: MODE_OPTIONS[0].label, // 'Matches'
-			value: '',
-			minValue: '',
-			maxValue: '',
+			filter: getDefaultValue(),
 		};
 	},
 
 	componentDidMount () {
 		// focus the text input
-		ReactDOM.findDOMNode(this.refs.input).focus();
+		findDOMNode(this.refs.focusTarget).focus();
```

Comment:
:+1: to all of that. it really bothers me that this stuff all got caught up with this PR though :(

I'm really tempted to merge into master and fix there before they get any more divergent, since this is already kind of a mess of a PR